src/main/java/nl/knaw/dans/vaultingest/core/DepositToBagProcess.java

```diff
@@ -99,6 +99,12 @@ void processDeposit(Deposit deposit) throws InvalidDepositException, IOException
             deposit.setNbn(idMinter.mintUrnNbn());
         }
 
+        var registeredDeposit = vaultCatalogService.registerDeposit(deposit);
+
+        if (registeredDeposit != null) {
+            deposit.setObjectVersion(registeredDeposit.getObjectVersion());
```

Comment:
Wasn't it the intention to have bags registered before storing them into the vault, to prevent having stored something without a trace? It looks like you continue without registration.

Reply:
Good point, it should just fail